oamapps/postConfigure/postConfigure.cpp

```diff
@@ -829,134 +834,147 @@ int main(int argc, char* argv[])
+					}
 
-			//check if dbrm data resides in older directory path and inform user if it does
-			dbrmDirCheck();
+					//set system DBRoot count and check 'files per parition' with number of dbroots
+					try
+					{
+						sysConfig->setConfig(SystemSection, "DBRootCount", oam.itoa(DBRootCount));
+					}
+					catch (...)
+					{
+						cout << "ERROR: Problem setting DBRoot Count in the MariaDB ColumnStore System Configuration file" << endl;
+						exit(1);
+					}
 
-			if (startOfflinePrompt)
-				offLineAppCheck();
-		}
+					//check if dbrm data resides in older directory path and inform user if it does
+					dbrmDirCheck();
 
-		checkMysqlPort(mysqlPort, sysConfig);
+					if (startOfflinePrompt)
+						offLineAppCheck();
 
-		if (hdfs || !rootUser)
-			if( !updateBash() )
-				cout << "updateBash error" << endl;
+					checkMysqlPort(mysqlPort, sysConfig);
 
-		if ( !writeConfig(sysConfig) ) {
-			cout << "ERROR: Failed trying to update MariaDB ColumnStore System Configuration file" << endl;
-			exit(1);
-		}
+					if ( !writeConfig(sysConfig) )
+					{
+						cout << "ERROR: Failed trying to update MariaDB ColumnStore System Configuration file" << endl;
+						exit(1);
+					}
 
-		cout << endl << "===== Performing Configuration Setup and MariaDB ColumnStore Startup =====" << endl;
+					cout << endl << "===== Performing Configuration Setup and MariaDB ColumnStore Startup =====" << endl;
 
-		cmd = installDir + "/bin/installer dummy.rpm dummy.rpm dummy.rpm dummy.rpm dummy.rpm initial dummy " + reuseConfig + " --nodeps ' ' 1 " + installDir;
-		system(cmd.c_str());
-		exit(0);
-	}
+					cmd = installDir + "/bin/installer dummy.rpm dummy.rpm dummy.rpm dummy.rpm dummy.rpm initial dummy " + reuseConfig + " --nodeps ' ' 1 " + installDir;
+					system(cmd.c_str());
+					exit(0);
+				}
+				else
+				{
+					if (temp == "2")
+					{
+						singleServerInstall = temp;
+						break;
+					}
+				}
 
-	try {
-		sysConfig->setConfig(InstallSection, "SingleServerInstall", "n");
-	}
-	catch(...)
-	{
-		cout << "ERROR: Problem setting SingleServerInstall from the MariaDB ColumnStore System Configuration file" << endl;
-		exit(1);
-	}
+				cout << "Invalid Entry, please re-enter" << endl;
 
-	if ( !writeConfig(sysConfig) ) {
-		cout << "ERROR: Failed trying to update MariaDB ColumnStore System Configuration file" << endl;
-		exit(1);
+				if ( noPrompting )
+					exit(1);
+
+				continue;
+			}
+
+        break;
+		}
 	}
 
-	//
-	// Multi-server install
-	//
+    try
+    {
+        sysConfig->setConfig(InstallSection, "SingleServerInstall", "n");
+    }
+    catch (...)
+    {
+        cout << "ERROR: Problem setting SingleServerInstall from the MariaDB ColumnStore System Configuration file" << endl;
+        exit(1);
+    }
+
+    if ( !writeConfig(sysConfig) )
+    {
+        cout << "ERROR: Failed trying to update MariaDB ColumnStore System Configuration file" << endl;
+        exit(1);
+    }
+
+    //
+    // Multi-server install
+    //
```

Comment:
these look like pretty significant changes that don't seem related to mcol-520.  Since it's config file related Daniel's tests should be able to expose problems.